src/harness/testcases/WINNF_FT_S_WDB_testcase.py

```diff
@@ -73,93 +180,92 @@ def generate_WDB_1_default_config(self, filename):
 
   @configurable_testcase(generate_WDB_1_default_config)
   def test_WINNF_FT_S_WDB_1(self, config_filename):
     """PAL Database Update."""
 
+    # Load the configuration file
     config = loadConfig(config_filename)
 
-    # Step 1: Register CBSD C to the SAS UUT.
-    cbsd_ids = self.assertRegistered([config['registrationRequest']],
-                        config['conditionalRegistrationData'])
-
-    # Step 2: Request creation of PPA 
-    pal_id = config['palRecord']['palId']
-    ppa_creation_request = {
-            "cbsdIds": cbsd_ids,
-            "palIds": [pal_id]
-        }
-    ppa_id = self._sas_admin.TriggerPpaCreation(ppa_creation_request)
+    # Step 1: Register device(s) 'C' with conditional registration data
+    cbsd_ids = self.assertRegistered(config['registrationRequests'],
+                                     config['conditionalRegistrationData'])
 
-    # Validate PPA creation request is failed.
-    self.assertEqual(ppa_id, None)
+    # Step 2: Request creation of PPA(s).
+    # Forming of PPA creation request(s)
+    ppa_creation_requests = []
+    for pal_ids in config['palIds']:
+      ppa_creation_requests.append({
+        "cbsdIds": cbsd_ids,
+        "palIds": pal_ids
+      })
 
-    # Step 3: Create PAL database with a PAL record containg the PAL ID P.
-    # Create fake PAL database server
-    fake_pal_database_server = FakeDatabaseTestHarness(
-                              config['fakePalDatabaseInfo']['databaseFile'],
-                              config['fakePalDatabaseInfo']['hostName'],
-                              config['fakePalDatabaseInfo']['port'])
+    # Trigger PPA creation
+    for ppa_creation_request in ppa_creation_requests:
+      self.assertPpaCreationFailure(ppa_creation_request)
 
-    # Start fake PAL database server
-    fake_pal_database_server.start()
-
-    # Inject the fake database URL into SAS UUT
-    database_url = fake_pal_database_server.getBaseUrl()
-    self._sas_admin.InjectDatabaseUrl(database_url)
+    # TODO
+    # Step 3: Create PAL database with a PAL record containg the PAL ID 'P'.
 
     # Step 4: Admin Test Harness triggers CPAS
     self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
 
-    # Step 4: Request PPA creation
-    ppa_id = self._sas_admin.TriggerPpaCreation(ppa_creation_request)
-    self.assertNotEqual(ppa_id, None)
-
-    # As Python garbage collector is not very consistent, the deletion
-    # of the fake_pal_database_server object is not immediate and consistent.
-    # Hence, explicitly stopping Database Test Harness and cleaning up.
-    fake_pal_database_server.shutdown()
-    del fake_pal_database_server
-  
+    # Step 5: Request creation of PPA(s) again
+    ppa_ids = []
+    for ppa_creation_request in ppa_creation_requests:
+      ppa_id = self.triggerPpaCreationAndWaitUntilComplete(ppa_creation_request)
+      logging.debug('ppa_id received from SAS UUT:%s', ppa_id)
+      ppa_ids.append(ppa_id)
+
+    del ppa_ids
```

Comment:
Why are you collecting a list of PPA IDs? And then deleting it?

Reply:
Thanks, 

Deleted   the unused IDs